frontend/src/components/userProfile/UserProfileForm.tsx

```diff
@@ -35,106 +37,191 @@ const UserProfileForm = ({
   setEmail,
   setEducationalQualification,
   setLanguageExperience,
-}: UserProfileFormProps) => (
-  <Flex direction="row" flex={1}>
-    <Flex direction="column" flex={1} maxWidth="50%">
-      <Heading size="lg">
-        {isSignup ? "Complete your profile!" : "Personal Information"}
-      </Heading>
-      {isSignup && (
-        <Text marginBottom="20px">
-          For the purposes of this platform, please enter your details
-        </Text>
-      )}
-      <FormControl>
-        <Heading marginTop="24px" size="sm">
-          Full name*
-        </Heading>
-        <Input
-          type="name"
-          value={fullName}
-          onChange={(e) => setFullName(e.target.value)}
-        />
-        <Heading marginTop="24px" size="sm">
-          Email*
-        </Heading>
-        <Input
-          type="email"
-          value={email}
-          onChange={(e) => setEmail(e.target.value)}
-        />
-        <Heading marginTop="24px" size="sm">
-          Educational qualifications*
-        </Heading>
-        <Textarea
-          height="150px"
-          type="qualifications"
-          onChange={(e) => setEducationalQualification(e.target.value)}
-        />
-        <Heading marginTop="24px" size="sm">
-          Language experience*
+  updateResume,
+}: UserProfileFormProps) => {
+  const [file, setFile] = useState<File | null>(null);
+
+  const onPreventDefault = (e: any) => {
+    e.preventDefault();
+    if (
+      e.dataTransfer.items &&
+      e.dataTransfer.items.length === 1 &&
+      e.dataTransfer.items[0].kind === "file"
+    ) {
+      e.dataTransfer.dropEffect = "copy";
+    }
+  };
+
+  const onFileDrop = (e: any) => {
+    e.preventDefault();
+    if (
+      e.dataTransfer.items &&
+      e.dataTransfer.items.length === 1 &&
+      e.dataTransfer.items[0].kind === "file"
+    ) {
+      const newResume = e.dataTransfer.items[0].getAsFile();
+      if (
+        newResume.type === "application/pdf" ||
+        newResume.type ===
+          "application/vnd.openxmlformats-officedocument.wordprocessingml.document"
+      ) {
+        setFile(newResume);
+        updateResume(newResume);
+      }
+    } else if (e.dataTransfer.files && e.dataTransfer.files.length === 1) {
+      const newResume = e.dataTransfer.files[0];
+      if (
+        newResume.type === "application/pdf" ||
+        newResume.type ===
+          "application/vnd.openxmlformats-officedocument.wordprocessingml.document"
+      ) {
+        setFile(newResume);
+        updateResume(newResume);
+      }
+    }
```

Comment:
This looks gnarly, but it is just checking that a single pdf or docx file is being dragged and dropped. I used both `e.dataTransfer.items` and `e.dataTransfer.files` for maximum browser support